app/src/main/java/com/teamcatchme/catchme/presentation/report/ReportDailyCharacterFragment.kt

```diff
@@ -52,6 +53,37 @@ class ReportDailyCharacterFragment : DialogFragment() {
         updateArrowVisibility()
     }
 
+    private fun setClickListener() {
+        binding.btnReportDialogMoveLeft.setOnClickListener { onArrowClicked(LEFT) }
+        binding.btnReportDialogMoveRight.setOnClickListener { onArrowClicked(RIGHT) }
+        binding.btnReportDialogClose.setOnClickListener { dismiss() }
+        binding.viewpagerReportDialog.registerOnPageChangeCallback(object :
+            ViewPager2.OnPageChangeCallback() {
+            override fun onPageScrolled(
+                position: Int,
+                positionOffset: Float,
+                positionOffsetPixels: Int
+            ) {
+                updateArrowVisibility()
+            }
+        })
+    }
+
+    private fun setDialogWidth() {
+        val catchmePixel = CatchMeApplication.pixelRatio
+        binding.constraintReportDialog.layoutParams.apply {
+            this.width = max(catchmePixel.screenWidth * 7 / 9, catchmePixel.toPixel(230))
+        }
+    }
+
+    private fun subscribeData() {
+        reportViewModel.characterList.observe(viewLifecycleOwner, {
+            val adapter = ReportDailyCharacterAdapter(it)
+            binding.viewpagerReportDialog.adapter = adapter
+            updateArrowVisibility()
+        })
```

Comment:
```suggestion
        reportViewModel.characterList.observe(viewLifecycleOwner) {
            val adapter = ReportDailyCharacterAdapter(it)
            binding.viewpagerReportDialog.adapter = adapter
            updateArrowVisibility()
        }
```

- 이렇게 바꿀 수 있음!
- 저러면 데이터가 바뀔때마다 어댑터가 생성될텐데 굳이 그럴 필요가 있을까?

Reply:
일회적으로 생각했었네 !!
이건 좀 열심히 고쳐서 다시 푸쉬할게욥